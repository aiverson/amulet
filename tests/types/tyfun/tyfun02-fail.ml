type x 'a =
  | Xi : x int
  | Xb : x bool
  | Xs : x string

type function collapse 'a begin
  collapse int = int
  collapse 'a = string
end

let collapse : forall 'a. x 'a -> x (collapse 'a) =
  function
  | Xi -> Xi
  | _ -> Xs
