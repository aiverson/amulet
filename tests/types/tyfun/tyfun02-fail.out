tyfun02-fail.ml[11:16 ..12:10]: error
   │ 
14 │   | _ -> Xs
   │          ^^
  No instance for collapse 'a ~ string arising from this expression
